lib/circuitbox/circuit_breaker.rb

```diff
@@ -52,15 +52,17 @@ def run!(run_options = {})
         logger.debug "[CIRCUIT] closed: querying #{service}"
 
         begin
-          t1 = Time.now
+          start_time = Time.now.to_i
           response = if exceptions.include? Timeout::Error
             timeout_seconds = run_options.fetch(:timeout_seconds) { option_value(:timeout_seconds) }
             timeout (timeout_seconds) { yield }
           else
             yield
           end
-          t2 = Time.now
-          log_circuit_execution_time(t2 - t1)
+          end_time = Time.now.to_i
+          if circuit_options[:notify_circuit_execution_time]
```

Comment:
While I like this to allow users to disable notifications for execution time if it's disabled we still actually collect timings. I like the idea of letting users specify their own timer class/object (we can default to ```Time```.

My thought being if I had a time class like the one below I could pass an object of this class to my circuit:

```ruby
class MonotonicTime
  def initialize(time_unit: :millisecond)
    @time_unit = time_unit
  end

  def now
    Process.clock_gettime(Process::CLOCK_MONOTONIC, @time_unit)
  end
end
```